openassessment/xblock/staff_area_mixin.py

```diff
@@ -323,26 +323,16 @@ def get_student_submission_context(self, student_username, submission):
         Returns:
             A context dict for rendering a student submission and associated rubric (for staff grading).
         """
-        if submission and 'file_key' in submission.get('answer', {}):
-            file_key = submission['answer']['file_key']
-
-            try:
-                submission['file_url'] = file_api.get_download_url(file_key)
-            except file_exceptions.FileUploadError:
-                # Log the error, but do not prevent the rest of the student info
-                # from being displayed.
-                msg = (
-                    u"Could not retrieve image URL for staff debug page.  "
-                    u"The learner username is '{student_username}', and the file key is {file_key}"
-                ).format(student_username=student_username, file_key=file_key)
-                logger.exception(msg)
-
         context = {
             'submission': create_submission_dict(submission, self.prompts) if submission else None,
             'rubric_criteria': copy.deepcopy(self.rubric_criteria_with_labels),
             'student_username': student_username,
         }
 
+        if submission:
+            context["file_upload_type"] = self.file_upload_type
+            context["staff_file_url"] = self.get_download_url_from_submission(submission)
```

Comment:
Using `self.get_download_url_from_submission(submission)` in place of the custom try block avoids duplicating that logic, at the cost of losing the specialized `student_username` logging. My thinking here is that the `for staff debug page.` line above indicates that this custom logging was more for debugging than usefulness.

The changed code now reflects more closely what the `peer_assessment_mixin` does: https://github.com/edx/edx-ora2/blob/master/openassessment/xblock/peer_assessment_mixin.py#L238